prow/git/publisher.go

```diff
@@ -0,0 +1,154 @@
+/*
```

Comment:
It seems a little strange to me that we are adding a separate git client for publishing. I feel like we should support any git operations on any repo. To me it seems like the actual distinction that needs to be handled differently is how we get a client for a repo. In scenario A we clone the repo and in scenario B we reference an existing local clone, but once we get a repo client shouldn't all the same operations be supported?